apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -129,8 +204,48 @@ const aichatSlice = createSlice({
         chatMessage.status = status;
       }
     },
-    setAiCustomizations: (state, action: PayloadAction<AiCustomizations>) => {
-      state.currentAiCustomizations = action.payload;
+    setStartingAiCustomizations: (
+      state,
+      action: PayloadAction<{
+        levelAiCustomizationsWithVisibility: LevelAiCustomizations;
+        studentAiCustomizations: AiCustomizations;
+      }>
+    ) => {
+      const {levelAiCustomizationsWithVisibility, studentAiCustomizations} =
+        action.payload;
+
+      let reconciledAiCustomizations: AiCustomizations = {
+        botName: levelAiCustomizationsWithVisibility.botName.value,
+        temperature: levelAiCustomizationsWithVisibility.temperature.value,
+        systemPrompt: levelAiCustomizationsWithVisibility.systemPrompt.value,
+        retrievalContexts:
+          levelAiCustomizationsWithVisibility.retrievalContexts.value,
+        modelCardInfo: levelAiCustomizationsWithVisibility.modelCardInfo.value,
+      };
+
+      for (const customizationUntyped in reconciledAiCustomizations) {
+        const customization = customizationUntyped as keyof AiCustomizations;
```

Comment:
This is also me battling Typescript :) Is there a better way to do this??

Reply:
Yeah unfortunately I don't know if there is (at least not that I know of) :/ I mean I think you could do something like

```
for (const customization of Object.keys(reconciledAiCustomizations) as (keyof AiCustomizations)[]) {
  // customization is keyof AiCustomization
  ... 
}
```
But don't know if that's much better than just casting it here